test/parallel/test-repl-delete-array-prototype.js

```diff
@@ -0,0 +1,10 @@
+'use strict';
+require('../common');
+const ArrayStream = require('../common/arraystream');
+const repl = require('repl');
+
+// Deleting Array.prototype.lastIndexOf used to immediately crash the REPL.
+// Refs: https://github.com/nodejs/node/pull/31457
+const input = new ArrayStream();
+repl.start({ input, output: process.stdout, useGlobal: true });
+input.run(['delete Array.prototype.lastIndexOf']);
```

Comment:
maybe we could do `Reflect.ownKeys(Array.prototype).forEach(x => delete Array.prototype[x])`? That's how I caught a number of the others I fixed here.

Reply:
turns out changing the test exposed even more issues :-) i'll add another commit shortly